api/api.go

```diff
@@ -769,8 +769,10 @@ func (c *Client) query(endpoint string, out interface{}, q *QueryOptions) (*Quer
 	parseQueryMeta(resp, qm)
 	qm.RequestTime = rtt
 
-	if err := decodeBody(resp, out); err != nil {
-		return nil, err
+	if out != nil {
+		if err := decodeBody(resp, out); err != nil {
```

Comment:
This allows us to use the `nil` to signal query that we expect an empty body. I'd prefer to change decodeBody to read `resp.ContentLength` to avoid parsing an empty response. Does that make sense?